### PR TITLE
integration with cake.cli

### PR DESCRIPTION
**Pull request by pitermarx**
@devlead

### CODE REVIEW

File: Cake.Console/Cake.Console.csproj [hunk 1]
```diff
@@ -8,6 +8,7 @@
     <PackageReference Include="cake.core" Version="1.2.0" />
     <PackageReference Include="cake.common" Version="1.2.0" />
     <PackageReference Include="cake.Nuget" Version="1.2.0" />
+    <PackageReference Include="cake.cli" Version="1.2.0" />
```
Comment (devlead) [inline Cake.Console/Cake.Console.csproj:11]: Package id has casing (NuGet normalizes so doesn't really matter)
```suggestion
    <PackageReference Include="Cake.Cli" Version="1.2.0" />

```

Reference example from NuGet
https://www.nuget.org/packages/Cake.CLI

File: Cake.Console/Internals/CakeContainer.cs [hunk 2]
```diff
@@ -42,7 +43,12 @@ public CakeContainer(CakeConsoleArguments args)
             services.Add(Builder.Singleton<IHostBuilderBehaviour, TaskRegisteringBehaviour>());
 
             // the host
-            services.Add(Builder.Singleton<IScriptHost, CakeHost>());
+            var host = args.HasArgument("dryrun") ? typeof(DryRunScriptHost) :
+                args.HasArgument("description") ? typeof(DescriptionScriptHost) :
+                args.HasArgument("tree") ? typeof(TreeScriptHost) :
+                typeof(CakeHost);
```
Comment (devlead) [inline Cake.Console/Internals/CakeContainer.cs:54]: Maybe an switch expression is more readable here?
```suggestion
            var host =
                (
                    DryRunScriptHost: args.HasArgument("dryrun"),
                    DescriptionScriptHost: args.HasArgument("description"),
                    TreeScriptHost: args.HasArgument("tree")
                ) switch {
                    { DryRunScriptHost: true }      => typeof(DryRunScriptHost),
                    { DescriptionScriptHost: true } => typeof(DescriptionScriptHost),
                    { TreeScriptHost: true }        => typeof(TreeScriptHost),
                    _                               => typeof(CakeHost)
                };
```

Comment (pitermarx) [reply Cake.Console/Internals/CakeContainer.cs:54]: In all candor, I disagree

Comment (devlead) [reply Cake.Console/Internals/CakeContainer.cs:54]: Totally ok :)

**Pull request merged**